portfolio/src/main/java/com/google/sps/data/BasketballCourtsDatastore.java

```diff
@@ -0,0 +1,102 @@
+package com.google.sps.data;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonElement;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import com.google.sps.data.BasketballCourt;
+
+/** Parses the basketball courts file and returns a list of basketball courts */
+public final class BasketballCourtsDatastore {
+  private static final Logger log = Logger.getLogger(BasketballCourtsDatastore.class.getName());
+
+  public static final String JSON_FILE_PATH =
+    "///home/riyanar/step/portfolio/src/main/webapp/WEB-INF/basketball-courts-nyc.json";
+  public static final String ACCESSIBLE_PROPERTY = "Accessible";
+  public static final String IS_ACCESSIBLE = "Y";
+  public static final String ID_PROPERTY = "Prop_ID";
+  public static final String NAME_PROPERTY = "Name";
+  public static final String LOCATION_PROPERTY = "Location";
+  public static final String NUMCOURTS_PROPERTY = "Num_of_Courts";
+  public static final String LAT_PROPERTY = "lat";
+  public static final String LON_PROPERTY = "lon";
+
+  private BasketballCourtsDatastore() {}
+
+  /** Returns a list of the accessible basketball courts. If there is a bad entry, it should be
+    ignored and the other entries should be returned. */
+  public static List<BasketballCourt> getAccessibleBasketballCourts() {
```

Comment:
Your cache should be in this class instead of your servlet. You can just keep track of your immutable list as a static value and potentially synchronize this method something like:

```
if( accessibleCourts != null ){ return accessibleCourts; }
synchronized(this){
   if( accessibleCourts != null ){ return accessibleCourts; }
   accessibleCourts = [...] load the list
}
return accessibleCourts;
```

It's ok to not synchronize it just means that multiple requests coming in at the same time when your value is not initialized yet will load it from file. not super optimized but works.